downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,557 @@
+
+@graphql:ServiceConfigurration {maxQueryDepth: 3}
+ Service /graphql on graphqlListener {
+    // Service definition
+}
+```
+
+##### WebSub Module Improvements
+
+Included functionality to the `websub:SubscriberService` to respond with user-defined custom payloads/header parameters in error scenarios.
+
+```ballerina
+import ballerina/websub;
+
+listener websub:Listener subscriberListener = new (9001);
+
+service /subscriber on subscriberListener {
+    remote function onSubscriptionValidationDenied(websub:SubscriptionDeniedError msg) returns websub:Acknowledgement? {
+        websub:Acknowledgement ack = {
+            headers = {
+                        "Content-Encoding" : "gzip"
+                    },
+            body = {
+                     "message" : "Successfully processed request"
+                 }
+        };
+        return ack;
+    }
+
+    remote function onSubscriptionVerification(websub:SubscriptionVerification msg) returns 
+    websub:SubscriptionVerificationSuccess|websub:SubscriptionVerificationError {
+        if (msg.hubTopic == "https://www.sample.topic") {
+            return error websub:SubscriptionVerificationError("Hub topic not supported", headers = 
+            {"Content-Encoding": "gzip"}, body = {"message": "Hub topic not supported"});
+        } else {
+            return {};
+        }
+    }
+
+    remote function onEventNotification(websub:ContentDistributionMessage event) returns websub:Acknowledgement|
+    websub:SubscriptionDeletedError? {
+        return error websub:SubscriptionDeletedError("Subscriber wants to unsubscribe", headers = 
+        {"Content-Encoding": "gzip"}, body = {"message": "Unsubscribing from the topic"});
+    }
+}
+```
+
+##### WebSubHub Module Improvements
+
+Included functionality to the `websubhub:Service` to respond with user-defined custom payloads/header parameters in error scenarios.
+
+```ballerina
+import ballerina / websubhub;
+
+service /websubhub on websubhub:Listener(9091) {
+
+    remote function onRegisterTopic(websubhub:TopicRegistration message) returns websubhub:TopicRegistrationSuccess|
+    websubhub:TopicRegistrationError {
+        if (message.topic == "https://sub.topic.com") {
+            websubhub:TopicRegistrationSuccess successResult = {body: <map<string>>{isSuccess: "true"}};
+            return successResult;
+        } else {
+            return error websubhub:TopicRegistrationError("Topic registration failed!", headers = 
+            {"Content-Encoding": "gzip"}, body = {"hub.additional.details": "Feature is not supported in the hub"});
+        }
+    }
+
+}
+```
+
+##### IO Module Improvements
+
+Introduce a parameter of type `XmlWriteOptions` to specify the entity type and the document type declaration.
+
+```ballerina
+public function fileWriteXml(@untainted string path, xml content, *XmlWriteOptions xmlOptions, FileWriteOption fileWriteOption = 
+                             OVERWRITE) returns Error? {
+}
+```
+
```

Comment:
```suggestion
**Example**

```ballerina
import ballerina/io;

public function main() returns error? {
    xml content = xml `<note>
        <to>Tove</to>
        <from>Jani</from>
        <heading>Reminder</heading>
        <body>Don't forget me this weekend!</body>
        </note>`;
    string publicId = "-//W3C//DTD HTML 4.01 Transitional//EN";
    string systemId = "http://www.w3.org/TR/html4/loose.dtd";
    var writeResult = io:fileWriteXml("./xmlFileWithDoc.xml", content, doctype = {
        system: systemId,
        'public: publicId
    });
}
```

**Output**
```xml
<!DOCTYPE note PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<note>
    <to>Tove</to>
    <from>Jani</from>
    <heading>Reminder</heading>
    <body>Don't forget me this weekend!</body>
</note>
```
```

Reply:
@MaryamZi @praneesha please approve.